circom/tests/simple/read_from_output.circom

```diff
@@ -0,0 +1,37 @@
+// REQUIRES: circom
+// RUN: rm -rf %t && mkdir %t && %circom --llzk -o %t %s | sed -n 's/.*Written successfully:.* \(.*\)/\1/p' | xargs cat | FileCheck %s --enable-var-scope
+// END.
+
+pragma circom 2.0.0;
+
+template ReadFromOutput() {
+    signal input inp;
+    signal output outp;
+    signal intermediate;
+    outp <== inp;
+    intermediate <== outp;
+}
+
+component main = ReadFromOutput();
+
+// CHECK-LABEL: module attributes {veridise.lang = "llzk"} {
+// CHECK-NEXT:   struct.def @ReadFromOutput<[]> {
+// CHECK-NEXT:     struct.field @outp : !felt.type {llzk.pub}
+// CHECK-NEXT:     struct.field @intermediate : !felt.type
+// CHECK-NEXT:     function.def @compute(%[[VAL_0:.*]]: !felt.type) -> !struct.type<@ReadFromOutput<[]>> attributes {function.allow_witness} {
+// CHECK-NEXT:       %[[VAL_1:.*]] = struct.new : <@ReadFromOutput<[]>>
+// CHECK-NEXT:       %[[VAL_2:.*]] = undef.undef : !felt.type
+// CHECK-NEXT:       struct.writef %[[VAL_1]][@outp] = %[[VAL_0]] : <@ReadFromOutput<[]>>, !felt.type
+// CHECK-NEXT:       struct.writef %[[VAL_1]][@intermediate] = %[[VAL_2]] : <@ReadFromOutput<[]>>, !felt.type
```

Comment:
There's something wrong in the code generation here. I verfied that circom generates the C code you would expect for the example:
```
PFrElement aux_dest = &signalValues[mySignalStart + 0];
Fr_copy(aux_dest,&signalValues[mySignalStart + 1]);
PFrElement aux_dest = &signalValues[mySignalStart + 2];
Fr_copy(aux_dest,&signalValues[mySignalStart + 0]);
```
which makes sense with the layout [`outp`, `inp`, `intermediate`].

Reply:
Yeah, I see the `AssignSignal` case in `template.rs` calls `set_named_value()` but the `AssignConstraintSignal` case does not.

Reply:
I'll submit a separate PR to fix